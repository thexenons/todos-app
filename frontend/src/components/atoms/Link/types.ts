import { PropsWithChildren } from "react";
import { PageKey } from "../../../pages";

export interface LinkProps extends PropsWithChildren {
	pageKey: PageKey;
}
