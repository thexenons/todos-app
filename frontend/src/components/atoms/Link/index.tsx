import type { FC } from "react";
import { Link as RouterLink } from "react-router-dom";
import { getPagePath } from "../../../router/utils";
import type { LinkProps } from "./types";

const Link: FC<LinkProps> = ({ pageKey, children }) => (
	<RouterLink to={getPagePath(pageKey)}>{children}</RouterLink>
);

export * from "./types";
export default Link;
