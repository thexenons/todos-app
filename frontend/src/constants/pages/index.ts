import { lazy } from "react";

import { PageKey, Pages } from "../../types";

export const PAGES: Pages = {
	[PageKey.home]: {
		path: "/",
		title: "Home",
		component: lazy(() => import("../../pages/Home")),
	},
	[PageKey.login]: {
		path: "/login",
		component: lazy(() => import("../../pages/Login")),
	},
	[PageKey.register]: {
		path: "/register",
		component: lazy(() => import("../../pages/Register")),
	},

	[PageKey.test]: {
		path: "/test",
		title: "Test",
		loader: async () => {
			return await (await import("../../pages/Test")).getInitialData();
		},
		component: lazy(() => import("../../pages/Test")),
		isProtected: true,
		children: {
			[PageKey.todosListSingle]: {
				path: "/:id",
				title: "Todos List Single",
				loader: async (args) => {
					return await (
						await import("../../pages/Test/TodosListSingle")
					).getInitialData(args);
				},
				component: lazy(() => import("../../pages/Test/TodosListSingle")),
			},
		},
	},
	[PageKey.components]: {
		path: "/components",
		title: "Components",
		component: lazy(() => import("../../pages/Components")),
		hideLayout: true,
		children: {
			[PageKey.componentsButton]: {
				path: "/button",
				title: "Button",
				component: lazy(() => import("../../pages/Components/Button")),
			},
			[PageKey.componentsAcceptButton]: {
				path: "/accept-button",
				title: "AcceptButton",
				component: lazy(() => import("../../pages/Components/AcceptButton")),
			},
			[PageKey.componentsUpdateButton]: {
				path: "/update-button",
				title: "UpdateButton",
				component: lazy(() => import("../../pages/Components/UpdateButton")),
			},
			[PageKey.componentsCreateButton]: {
				path: "/create-button",
				title: "CreateButton",
				component: lazy(() => import("../../pages/Components/CreateButton")),
			},
			[PageKey.componentsDeleteButton]: {
				path: "/delete-button",
				title: "DeleteButton",
				component: lazy(() => import("../../pages/Components/DeleteButton")),
			},
			[PageKey.componentsCancelButton]: {
				path: "/cancel-button",
				title: "CancelButton",
				component: lazy(() => import("../../pages/Components/CancelButton")),
			},
			[PageKey.componentsPanel]: {
				path: "/panel",
				title: "Panel",
				component: lazy(() => import("../../pages/Components/Panel")),
			},
			[PageKey.componentsModal]: {
				path: "/modal",
				title: "Modal",
				component: lazy(() => import("../../pages/Components/Modal")),
			},
			[PageKey.componentsFloatingPanel]: {
				path: "/floating-panel",
				title: "FloatingPanel",
				component: lazy(() => import("../../pages/Components/FloatingPanel")),
			},
			[PageKey.componentsInputBase]: {
				path: "/input-base",
				title: "InputBase",
				component: lazy(() => import("../../pages/Components/InputBase")),
			},
		},
	},
};
