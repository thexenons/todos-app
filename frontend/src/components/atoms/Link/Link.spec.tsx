import { render } from "@testing-library/react";

import { PageKey } from "../../../constants";
import BaseTestComponent from "../../../tests/BaseTestComponent";
import Link from ".";

describe("Link", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<Link pageKey={PageKey.home} />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
