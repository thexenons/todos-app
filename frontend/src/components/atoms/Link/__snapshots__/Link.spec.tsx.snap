// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > Render 1`] = `
<div>
  <a
    href="/"
  />
</div>
`;
